modyn/trainer_server/internal/grpc/trainer_server_grpc_servicer.py

```diff
@@ -75,12 +100,39 @@ def start_training(
             return StartTrainingResponse(training_started=False)
 
         pretrained_model_path: Optional[pathlib.Path] = None
-        if request.pretrained_model_path is not None and request.pretrained_model_path != "":
-            pretrained_model_path = self._modyn_base_dir / pathlib.Path(request.pretrained_model_path)
-            if not pretrained_model_path.exists():
-                logger.error(f"Pretrained Model Path {pretrained_model_path} does not exist. Cannot start training.")
+        if request.use_pretrained_model and request.pretrained_model_id is not None:
+            fetch_request = FetchModelRequest(model_id=request.pretrained_model_id)
+            fetch_resp: FetchModelResponse = self.model_storage_stub.FetchModel(fetch_request)
+
+            if not fetch_resp.success:
+                logger.error(
+                    f"Pretrained Model {request.pretrained_model_id} cannot be fetched from model storage. "
+                    f"Training cannot be started."
+                )
                 return StartTrainingResponse(training_started=False)
 
+            last_progress = 0.0
+
+            def download_callback(current_progress: float) -> None:
+                nonlocal last_progress
+                for emit_perc in EMIT_MESSAGE_PERCENTAGES:
+                    if last_progress <= emit_perc < current_progress:
+                        logger.info(f"Completed {emit_perc * 100}% of the pretrained model download.")
+                last_progress = current_progress
+
+            pretrained_model_path = self._modyn_base_dir / pathlib.Path(f"model_{request.pretrained_model_id}.modyn")
+            download_file(
+                hostname=self._config["model_storage"]["hostname"],
+                port=int(self._config["model_storage"]["ftp_port"]),
+                user="modyn",
+                password="modyn",
+                remote_file_path=pathlib.Path(fetch_resp.model_path),
+                local_file_path=pretrained_model_path,
+                callback=download_callback,
+            )
+
+            logger.info("Completed pretrained model download.")
```

Comment:
let's put this in a separate function, like _download_pretrained_model

Reply:
Done